examples/svelte/basic/src/App.svelte

```diff
@@ -0,0 +1,148 @@
+<script lang="ts">
+  import './index.css'
+  import {
+    createTable,
+    getCoreRowModel,
+    createTableInstance,
+  } from '@tanstack/svelte-table'
+  import { writable } from 'svelte/store'
+
+  type Person = {
+    firstName: string
+    lastName: string
+    age: number
+    visits: number
+    status: string
+    progress: number
+  }
+
+  const table = createTable().setRowType<Person>()
+
+  const defaultData: Person[] = [
+    {
+      firstName: 'tanner',
+      lastName: 'linsley',
+      age: 24,
+      visits: 100,
+      status: 'In Relationship',
+      progress: 50,
+    },
+    {
+      firstName: 'tandy',
+      lastName: 'miller',
+      age: 40,
+      visits: 40,
+      status: 'Single',
+      progress: 80,
+    },
+    {
+      firstName: 'joe',
+      lastName: 'dirte',
+      age: 45,
+      visits: 20,
+      status: 'Complicated',
+      progress: 10,
+    },
+  ]
+
+  const defaultColumns = [
+    table.createGroup({
+      header: 'Name',
+      footer: props => props.column.id,
+      columns: [
+        table.createDataColumn('firstName', {
+          cell: info => info.getValue(),
+          footer: props => props.column.id,
+        }),
+        table.createDataColumn(row => row.lastName, {
+          id: 'lastName',
+          cell: info => info.getValue(),
+          header: () => 'Last Name',
+          footer: props => props.column.id,
+        }),
+      ],
+    }),
+    table.createGroup({
+      header: 'Info',
+      footer: props => props.column.id,
+      columns: [
+        table.createDataColumn('age', {
+          header: () => 'Age',
+          footer: props => props.column.id,
+        }),
+        table.createGroup({
+          header: 'More Info',
+          columns: [
+            table.createDataColumn('visits', {
+              header: () => 'Visits',
+              footer: props => props.column.id,
+            }),
+            table.createDataColumn('status', {
+              header: 'Status',
+              footer: props => props.column.id,
+            }),
+            table.createDataColumn('progress', {
+              header: 'Profile Progress',
+              footer: props => props.column.id,
+            }),
+          ],
+        }),
+      ],
+    }),
+  ]
+
+  let data = writable<Person[]>(defaultData)
+  let columns = writable(defaultColumns)
+
+  const rerender = () => {
+    data = data
+  }
+
+  const instance = createTableInstance(table, {
+    data: $data,
+    columns: $columns,
+    getCoreRowModel: getCoreRowModel(),
+  })
```

Comment:
I'm curious as to was this example rerendering for you on clicking the button? This was how the API was supposed to work for updatable options:-

```js
const options = writable({
  data: defaultData,
  columns: defaultColumns,
  getCoreRowModel: getCoreRowModel()
});

const rerender = () => {
  options.update(options => ({
    ...options,
    data: defaultData
  }));
}

const instance = createTableInstance(table, options);
```

Reply:
yep, that makes more sense. Will change to that

Reply:
after upgrading to the new alpha, I get this error

```Uncaught TypeError: switch_value is not a constructor```

Transpile error again?

Reply:
`renderCell` and `renderFooter` should be `renderCell()` and `renderFooter` in the template. I think that should make it work

Reply:
silly mistake on my part, fixed, but same error. :(

Reply:
This was a bug in the adapter that didn't allow cases where a function was passed as header rather than the string/number literal. I'll open a pr to fix it

Reply:
Thanks for working through this this week.

Reply:
#3917

Reply:
It is working identical to the react basic example now